### PR TITLE
Extend analysis_runner.py for backfilling versions for a given package

### PR DESCRIPTION
**Pull request by calebbrown**
This helps with experimentation/testing by having access to a range of versions for a given package.

### CODE REVIEW

File: tools/analysis/analysis_runner.py [hunk 1]
```diff
@@ -1,13 +1,51 @@
...
+  url = f'https://registry.npmjs.com/{safe_pkg}'
+  resp = urllib.request.urlopen(url)
+  data = json.loads(resp.read())
+  versions = data.get('time', {}).keys()
+  return [v for v in versions if v not in _NPM_IGNORE_KEYS][::-1]
+
+
+def _pypi_versions_for_package(pkg):
+  safe_pkg = urllib.parse.quote_plus(pkg)
+  url = f'https://pypi.org/rss/project/{safe_pkg}/releases.xml'
```
Comment (oliverchang) [inline tools/analysis/analysis_runner.py:29]: nit: The Python JSON API is much simpler and doesn't require XML:

e.g. OSV uses this for its own version enumeration:

https://github.com/google/osv/blob/737948b5060efeffb1cf47fe085d187a3d0304a4/lib/osv/ecosystems.py#L126

File: tools/analysis/analysis_runner.py [hunk 2]
```diff
@@ -71,9 +114,14 @@ def main():
             results_bucket=args.results)
 
   elif args.name:
-    _request(
-        args.name, args.ecosystem, args.version, local_file=args.file,
-        results_bucket=args.results)
+    if args.all:
```
Comment (oliverchang) [inline tools/analysis/analysis_runner.py:117]: nit: We might want to support --all for other ways ot specifying packages too. e.g. via "--list --all".

**Review by oliverchang — APPROVED** [review]
nice!

**Review by oliverchang — APPROVED** [review]
thanks!

File: tools/analysis/analysis_runner.py (hunk 2 above)
Comment (calebbrown) [reply tools/analysis/analysis_runner.py:120]: Good idea. Added support

File: tools/analysis/analysis_runner.py (hunk 1 above)
Comment (calebbrown) [reply tools/analysis/analysis_runner.py]: That API is much better. Thanks!

**Pull request merged**